modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/cards/HorizontalCard.js

```diff
@@ -37,6 +37,20 @@ export default function HorizontalCard({
 }) {
 	const [selected, setSelected] = useState(initialSelected);
 
+	const handleAllItemsToggled = ({checked}) => {
+		setSelected(checked);
+	};
+
+	useEffect(() => {
+		Liferay.on('allItemsToggled', handleAllItemsToggled);
+
+		return () => {
+			Liferay.detach('allItemsToggled', handleAllItemsToggled);
+		};
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
```

Comment:
We should avoid the lint suppression here. People override it so often that we collectively start to consider it to be "ok" to do so, but I think we should hold a higher bar.

So, add `handleAllItemsToggled` to the dependencies array, and use `useCallback` to ensure that the `handleAllItemsToggled` callback has a stable identity over time (the identity of `setSelected` itself will be stable, because React takes care to ensure that in its `useState` implementation).